src/utils/encryption.ts

```diff
@@ -1,83 +1,145 @@
-import * as crypto from 'crypto';
 import * as eciesjs from 'eciesjs';
+import { AesCtr } from '@web5/crypto';
+import type { Jwk } from '@web5/crypto';
 import { Readable } from 'readable-stream';
 
-// compress publicKey for message encryption
+// Compress publicKey for message encryption
 eciesjs.ECIES_CONFIG.isEphemeralKeyCompressed = true;
 
+export interface EciesEncryptionOutput {
+  ciphertext: Uint8Array;
+  ephemeralPublicKey: Uint8Array;
+  initializationVector: Uint8Array;
+  messageAuthenticationCode: Uint8Array;
+}
+
+export interface EciesEncryptionInput {
+  privateKey: Uint8Array;
+  ephemeralPublicKey: Uint8Array;
+  initializationVector: Uint8Array;
+  messageAuthenticationCode: Uint8Array;
+  ciphertext: Uint8Array;
+}
+
 /**
  * Utility class for performing common, non-DWN specific encryption operations.
  */
 export class Encryption {
   /**
-   * Encrypts the given plaintext stream using AES-256-CTR algorithm.
+   * Converts a key to base64url encoding
+   * @param key - Uint8Array to convert
    */
-  public static async aes256CtrEncrypt(key: Uint8Array, initializationVector: Uint8Array, plaintextStream: Readable): Promise<Readable> {
-    const cipher = crypto.createCipheriv('aes-256-ctr', key, initializationVector);
+  public static isEphemeralKeyCompressed: boolean = true; // Set default value
+
+  private static toBase64Url(buffer: Buffer): string {
+    return buffer.toString('base64') // Convert to base64
+      .replace(/\+/g, '-') // Replace + with -
+      .replace(/\//g, '_') // Replace / with _
+      .replace(/=+$/, ''); // Remove any trailing '='
+  }
+
+  private static async convertToJwk(key: Uint8Array): Promise<Jwk> {
+    return {
+      kty : 'oct',
+      k   : this.toBase64Url(Buffer.from(key)), // Use the new base64url method
+      alg : 'A256CTR',
+      ext : 'true',
+    };
+  }
```

Comment:
These should be able to be replaced with the use of the convenience function from the crypto library:
` AesCtr.bytesToPrivateKey()`